### ISSUE TITLE
warnings from emcee 2.1.0

### ISSUE THREAD

**Issue by newville** [lmfit/lmfit-py#350]
I get warnings from the nosetests with emcee 2.1.0:
```
......................................................S....S................................................................................../Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:278: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  self.dim))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:279: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  self._lnprob = np.zeros((self.ntemps, self.nwalkers, nsave))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:281: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  nsave))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:308: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  zs = np.exp(np.random.uniform(low=-np.log(self.a), high=np.log(self.a), size=(self.ntemps, self.nwalkers/2)))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:310: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  qs = np.zeros((self.ntemps, self.nwalkers/2, self.dim))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:313: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  size=self.nwalkers / 2)
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:315: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  (self.nwalkers / 2, 1)) * (pupdate[k, :, :] -
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:328: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  (self.ntemps, self.nwalkers/2))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:330: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  (self.ntemps, self.nwalkers/2))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:338: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  self.nwalkers/2)))
/Users/Newville/anaconda/lib/python2.7/site-packages/emcee/ptsampler.py:350: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  accepts = accepts.reshape((self.ntemps, self.nwalkers/2))
..........capi_return is NULL
Call-back cb_calcfc_in__cobyla__user__routines failed.

```

Anyone know why?

Possibly related:  I can't run `examples\lmfit_and_emcee.py`, it gives:

```
python  examples/lmfit_and_emcee.py 
[[Variables]]
    a:    5          (init= 5)
    b:   -5          (init=-5)
    t1:   1          (init= 1)
    t2:   15         (init= 15)
[[Correlations]] (unreported correlations are <  0.100)
Traceback (most recent call last):
  File "examples/lmfit_and_emcee.py", line 99, in <module>
    lnfunc, guess = create_all(mini)
  File "examples/lmfit_and_emcee.py", line 66, in create_all
    guess = starting_guess(mini, not sigma_given)
  File "examples/lmfit_and_emcee.py", line 52, in starting_guess
    vals.append(mini.residual.std())
AttributeError: 'NoneType' object has no attribute 'std'
```

**Comment by andyfaff** [comment]
I was experiencing a similar problem (not the ones you are reporting)  with the latest version on PyPi.  The issues arise due to changed numpy behaviour.  You now can't create a numpy array with non-ints, e.g. `np.zeros(num)` doesn't work if `num` is not an `int`.
I submitted a PR in https://github.com/dfm/emcee/pull/175, which got merged. To be fair most of those occurrences in `ptsampler.py` were addressed in https://github.com/dfm/emcee/commit/a4da655815a30011ec189d4d2f93a73d5b4fff98, but I got the one that was missed.
Unfortunately the last release of emcee on PyPi (2.1.0) was in May 2014, and the issues were fixed in emcee/master in August 2014. So the issues that we're seeing are not there if one uses emcee/master. 
I'll raise an issue on the emcee project and tell them that we are experiencing these problems which will only get fixed if they cut a new release on PyPi.

**Comment by newville** [comment]
@andyfaff sorrry for the confusion.  I do have emcee version 2.1.0 (installed with conda).

**Comment by andyfaff** [comment]
Could you get Travis to rerun the test suite? @dfm has released an updated version of emcee on pypi  which should fix these issues.

**Comment by newville** [comment]
@andyfaff yes, using 2.2.0 removes these warnings!

**Comment by newville** [comment]
seems resolved by using emcee 2.2.0.
